drivers/grove/light_sensor.c

```diff
@@ -71,15 +71,26 @@ static int gls_init(struct device *dev)
 		return -EINVAL;
 	}
 
-	drv_data->sample.sampling_delay = 12;
-	drv_data->sample.channel_id = CONFIG_GROVE_LIGHT_SENSOR_ADC_CHANNEL;
-	drv_data->sample.buffer = drv_data->adc_buffer;
-	drv_data->sample.buffer_length = 4;
-
-	drv_data->adc_table.entries = &drv_data->sample;
-	drv_data->adc_table.num_entries = 1;
-
-	adc_enable(drv_data->adc);
+#if defined(CONFIG_ARC)
```

Comment:
I do not see much difference here, just 1 entry?
Make the ifdef apply on to that different entry and avoid duplicating all other elements.

checking for architecture here (CONFIG_ARC) is wrong, you should be checking for the specific driver being used, an ARC system can have different drivers other than designware.